direct/io-kafka/src/main/java/cz/o2/proxima/direct/kafka/KafkaLogReader.java

```diff
@@ -392,14 +393,17 @@ private void submitConsumerWithObserver(
                 log.debug(
                     "End offsets of current assignment {}: {}", kafka.assignment(), endOffsets);
               }
-
-              listener.onPartitionsAssigned(kafka.assignment());
             } while (poll.isEmpty()
+                && numPolls++ < 2
```

Comment:
can you please increment the counter inside the cycle, so the construct is more human readable?

Reply:
Hm, why is this not readable? I think this is pretty common pattern. I can do that, of course, but I don't think that will result in better readability.